stored_requests/config/config.go

```diff
@@ -46,8 +47,9 @@ func NewStoredRequests(cfg *config.Configuration, client *http.Client, router *h
 	eventProducers, ampEventProducers := newEventProducers(&cfg.StoredRequests, client, db, router)
 	cache := newCache(&cfg.StoredRequests)
 	ampCache := newCache(&cfg.StoredRequests)
-	fetcher, ampFetcher = newFetchers(&cfg.StoredRequests, client, db)
-	categoriesFetcher, _ = newFetchers(&cfg.CategoryMapping, client, db)
+	fetcher = newFetcher(&cfg.StoredRequests, client, db, false)
```

Comment:
I would suggest creating a new struct that supports only one fetcher, rather than the current `cfg.StoredRequests`, and copying the data from `cfg.StoredRequests` into two copies of the single fetcher struct. That way we can have a CategoryMapping config entry that does not include phantom AMP support. It should also allow you to get rid of the `isAmp` parameter.

Reply:
Or, alternatively, use the structure to hold all fetchers (Endpoint, AmpEndpoint, CategoryEndpoint) but change _NewFetcher_()'s **isAmp** argument to be a pointer to the desired _cfg.HTTP.<whatever>Endpoint_.